extensions/amp-next-page/0.1/amp-next-page.js

```diff
@@ -270,8 +271,16 @@ export class AmpNextPage extends AMP.BaseElement {
     });
 
     const docInfo = Services.documentInfoForDoc(this.element);
-    const host = parseUrl(docInfo.sourceUrl).host;
-    this.config_ = assertConfig(configJson, host);
+    const url = parseUrl(docInfo.sourceUrl);
+
+    this.config_ = assertConfig(configJson, url.host, getSourceOrigin(url));
+
+    if (isProxyOrigin(url)) {
+      const sourceOrigin = getSourceOrigin(url);
+      configJson.pages.forEach(rec => {
```

Comment:
should become `configJson['pages']` and handle the edge case where it may be null or not be an array.

Reply:
Oops, should've been this.config_.pages.

null/isArray checks already done in `assertConfig`.